app/api.js

```diff
@@ -22,9 +24,22 @@ export function publishSiteState({ community, conference, events }) {
     },
     body,
   })
-  .then(response => {
-    if (response.status === 201) return response.json();
-    throw new Error('Unable to publush site content');
+  .then(assertStatusOK)
+  .then(parseJSON);
+}
+
+function get(url) {
+  return fetch(url, {
+    credentials: 'same-origin',
   })
-  .catch(error => { throw error; });
+  .then(assertStatusOK)
+  .then(parseJSON);
+}
+
+export function fetchSiteState() {
+  return get(SITE_ENDPOINT);
+}
+
+export function publishSiteState({ community, conference, events }) {
```

Comment:
I want to filter out any other properties so I know the shape of the data that I'm passing here. It also makes assertions about the data type as this will crash if it's not an object. Perhaps this is not so much of a Javascript-y thing to do